palette/moreland/convert.go

```diff
@@ -0,0 +1,229 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package moreland
+
+import (
+	"image/color"
+	"math"
+)
+
+// rgb repesents a physically linear RGB color.
+type rgb struct {
+	R, G, B float64
+}
+
+// cieXYZ converts a linear RGB color to a CIE XYZ color.
+func (c rgb) cieXYZ() cieXYZ {
+	return cieXYZ{
+		X: 0.4124*c.R + 0.3576*c.G + 0.1805*c.B,
+		Y: 0.2126*c.R + 0.7152*c.G + 0.0722*c.B,
+		Z: 0.0193*c.R + 0.1192*c.G + 0.9505*c.B,
+	}
+}
+
+// sRGBA converts a linear RGB color to an sRGBA color, where alpha is the alpha
+// channel value (between 0 and 1).
+func (c rgb) sRGBA(alpha float64) sRGBA {
+	return sRGBA{
+		R: linearToS(c.R),
+		G: linearToS(c.G),
+		B: linearToS(c.B),
+		A: alpha,
+	}
+}
+
+// linearToS converts from a linear RGB component to an sRGB component.
+func linearToS(v float64) float64 {
+	if v > 0.0031308 {
+		return 1.055*math.Pow(v, 1/2.4) - 0.055
+	}
+	return 12.92 * v
+}
+
+// cieXYZ represents a color in CIE XYZ space.
+type cieXYZ struct {
+	X, Y, Z float64
```

Comment:
Please add an annotation explaining the appropriate ranges of these fields' values.